src/digest.rs

```diff
@@ -448,8 +447,66 @@ pub extern fn SHA512_4(out: *mut u8, out_len: c::size_t,
     polyfill::slice::fill_from_slice(out, digest);
 }
 
+unsafe extern fn sha1_block_data_order(state: *mut u64, data: *const u8, num: c::size_t) {
+    sha1_process_blocks(
+        &mut *(state as *mut _),
+        core::slice::from_raw_parts(data as *const _, num)
```

Comment:
This cast is not valid because there's no guarantee that `data` is aligned enough for accessing as `u32`. This is why the C code uses `from_be_u32_ptr`.